service/collector/tiller_configured.go

```diff
@@ -0,0 +1,90 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	tillerConfiguredDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "tiller_configured"),
+		"Tiller is configured correctly.",
+		[]string{
+			namespaceLabel,
+		},
+		nil,
+	)
+)
+
+func (c *Collector) collectTillerConfigured(ctx context.Context, ch chan<- prometheus.Metric) {
+	var value float64
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "collecting Tiller configuration")
+
+	charts, err := c.getCharts()
+	if err != nil {
+		c.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("could not get Charts"), "stack", fmt.Sprintf("%#v", err))
+		return
+	}
+
+	chartConfigs, err := c.getChartConfigs()
+	if err != nil {
+		c.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("could not get ChartConfigs"), "stack", fmt.Sprintf("%#v", err))
+		return
+	}
+
+	if len(charts) == 0 && len(chartConfigs) == 0 {
+		// Skip checking tiller when there are no custom resources,
+		// as tiller is only installed when there is at least one CR to reconcile.
+		c.logger.Log("level", "error", "message", "did not collect Tiller configuration")
+		c.logger.Log("level", "error", "message", "no ChartConfg CRs in the cluster")
+
+		value = 1
+	} else {
+		err := c.checkTillerDeployment()
+		if err != nil {
+			c.logger.Log("level", "error", "message", "failed to collect Tiller configuration", "stack", fmt.Sprintf("%#v", err))
+
+			value = 0
+		} else {
+			value = 1
+		}
+	}
+
+	ch <- prometheus.MustNewConstMetric(
+		tillerConfiguredDesc,
+		prometheus.GaugeValue,
+		value,
+		c.tillerNamespace,
+	)
+
+	c.logger.LogCtx(ctx, "level", "debug", "message", "finished collecting Tiller configuration")
+}
+
+func (c *Collector) checkTillerDeployment() error {
+	deploy, err := c.k8sClient.Extensions().Deployments(c.tillerNamespace).Get("tiller-deploy", metav1.GetOptions{})
```

Comment:
Does it make sense to put `"tiller-deploy"` into `key` package?

Reply:
Yes good shout.